library/src/main/android/permissions/dispatcher/PermissionUtils.java

```diff
@@ -94,13 +96,23 @@ public static boolean hasSelfPermissions(Context context, String... permissions)
      * @see #hasSelfPermissions(Context, String...)
      */
     private static boolean hasSelfPermission(Context context, String permission) {
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.M && Build.MANUFACTURER.equalsIgnoreCase("Xiaomi")) {
```

Comment:
Is there any other way to detect a Xiaomi device than to use `Build.MANUFACTURER` here? I just want to make sure we can catch all of them, and that there aren't any weird devices with an identifier like `xia` or something.